packages/logger-plugin/src/action-logger.ts

```diff
@@ -0,0 +1,45 @@
+import { getActionTypeFromInstance, Store } from '@ngxs/store';
+
+import { formatTime } from './internals';
+import { LogWriter } from './log-writer';
+
+export class ActionLogger {
+  constructor(private action: any, private store: Store, private logWriter: LogWriter) {}
+
+  dispatched(state: any) {
+    const actionName = getActionTypeFromInstance(this.action);
+    const formattedTime = formatTime(new Date());
+
+    const message = `action ${actionName} @ ${formattedTime}`;
+    this.logWriter.startGroup(message);
+
+    // print payload only if at least one property is supplied
+    if (this._hasPayload(this.action)) {
+      this.logWriter.logGrey('payload', { ...this.action });
+    }
+
+    this.logWriter.logGrey('prev state', state);
```

Comment:
what is 'prev state', 'next state', 'next state after error', 'error' ?

Maybe we usage 

```ts
export enum REDUX_LOG_PREFIX {
 PREV = 'prev state',
 NEXT = 'next state',
 ERROR = 'error',
 NEXT_AFTER_ERROR = 'next state after error'
}
```

or something

Reply:
But if you want to write tests, you should use them to compare

Reply:
Your tests should not use production code to validate production code.
The test should be explicit about its expectations. If the production code changes behaviour then the test should fail. Just a small tip from our TDD 101 course

Reply:
Left it as is.